### PR TITLE
866 Reading and Writing to the Blockchain

### LINKED ISSUE #866: How to read/write data into the blockchain
Improve https://docs.casperlabs.io/design/reading-and-writing-to-the-blockchain/
- transform it into a “how to read/write data into the blockchain”

### PR DESCRIPTION
**Pull request by ACStoneCL**
### What does this PR fix/introduce?
Expanded information on the methods of storage for Casper networks.

Closes #866 

### Additional context
Includes work previously done by @ipopescu to refine the existing [document](https://docs.casperlabs.io/concepts/design/reading-and-writing-to-the-blockchain/) as well as a new document "Reading and Writing to Global State using Rust" that outlines more specifics.

### Checklist
(Delete any that aren't relevant)

- [x] Docs are successfully building - `yarn install && yarn run build`.
- [x] For new **internal** links I used *relative file paths* (with .md extension) - e.g. `../../faq/faq-general.md` - instead of introducing *absolute file path*, or *relative/absolute URL*.
- [x] All external links have been verified with `yarn run check:externals`.
- [x] My changes follow the [Casper docs style guidelines](https://docs.casperlabs.io/workflow/contribute/).

### Reviewers
Writing: @ipopescu / Tech: @gRoussac 
FYI: @darthsiroftardis

### CODE REVIEW

**Review by ipopescu — COMMENTED** [review]
Looks good, @ACStoneCL. Hoping I understood the code, I left some clarifications that might help. I will do another round once we resolve the comments.
The referenced [repository](https://github.com/casper-ecosystem/tutorials-example-wasm/tree/dev/storage-example) needs to become public; otherwise, the links in the documentation will not be found, and the docs build will fail. This is probably why the build is failing now.

File: source/docs/casper/resources/tutorials/advanced/storage-workflow.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+Once the stored value has been initialized, the `storage::write` function overwrites the existing value with `true`. The URef is then stored in the current context as a `NamedKey` titled `MY_STORED_VALUE_UREF`.
+
+```rust
+
+    // Store contract hash under a Named key CONTRACT_HASH
+    runtime::put_key(CONTRACT_HASH, contract_hash.into());
+
+    // Store !MY_STORED_VALUE (false) as init value/type into a new URef
+    let my_value_uref = storage::new_uref(!MY_STORED_VALUE);
```
Comment (gRoussac) [inline source/docs/casper/resources/tutorials/advanced/storage-workflow.md:37]: @darthsiroftardis I took this storage::new_uref to create a new uref, is there any other way that does not use a initial value ? thank you

**Review by gRoussac — APPROVED** [review]
@ACStoneCL thank you it seems clear to me. 

I just have a small doubt regarding usage of `new_uref`, I am wondering if there is another way than with an initial value, that would make the example maybe clearer.

**Review by ipopescu — APPROVED** [review]
Looks great!

**Comment by ACStoneCL** [comment]
> @ACStoneCL thank you it seems clear to me.
> 
> I just have a small doubt regarding usage of `new_uref`, I am wondering if there is another way than with an initial value, that would make the example maybe clearer.

I'll wait for now to see if there's an answer to this one.

**Pull request merged**